tempesta_fw/str.c

```diff
@@ -140,6 +140,11 @@ void tfw_str_collect_cmp(TfwStr *chunk, TfwStr *end, TfwStr *out,
 
 	BUG_ON(!TFW_STR_PLAIN(chunk));
 
+	if (unlikely(chunk == end)) {
+		bzero_fast(out, sizeof(*out));
+		return;
+	}
+
```

Comment:
This function was intended to collect cookie value in any case (or `BUG_ON` must be triggered), so it didn't return value. But now it can return with no value found and caller will not know anything about this. Perhaps it is worth to change it to return `true/false` or `0/-ENOENT`; also the similar checks before `tfw_str_collect_cmp` calls - looks redundant now.

Reply:
I can't shake the feeling that it's wrong to add any return value to this function. It has a verb in the name, so by convention should return 0 if there are no errors, and other values otherwise. But this function cannot fail. It's not an error to return empty string if an empty slice was provided. If we add a return value, somebody reading code at the call site will get an impression of error handling, which it is not.

So I think it is good to leave testing for `chunk == end` before calling this function. It's a duplication, but I don't think checking should be removed from this function, as it may result in accidental reading outside of allocated memory.